File: 1354-find-players-with-zero-or-one-losses/find-players-with-zero-or-one-losses.php
class Solution {

    /**
     * @param Integer[][] $matches
     * @return Integer[][]
     */
    function findWinners($matches) {
        $checker = [];
        foreach($matches as $match){
            $winner = $match[0];
            $loser = $match[1];
            // check if the winner is in checker 
            if(!isset($checker[$winner])){
                $checker[$winner] = 0;
            }
            // check if loser is in checker 
            if(!isset($checker[$loser])){
                $checker[$loser] = 0;
            }

            $checker[$loser] += 1;
        }

        $res = [[],[]];
        foreach ($checker as $player => $losses){
            if ($losses == 0){
                $res[0][] = $player;
            }elseif ($losses == 1){
                $res[1][] = $player;
            }

        }
        sort($res[1]);
        sort($res[0]);

        return $res;
    }
}

File: 1354-find-players-with-zero-or-one-losses/find-players-with-zero-or-one-losses.py
class Solution:
    def findWinners(self, matches: List[List[int]]) -> List[List[int]]:
        checker = {}
        for i in matches:
            checker[i[0]] = checker.get(i[0],0)
            checker[i[1]] = checker.get(i[1],0) + 1       
 
        res = [[],[]]
        for num, wins in checker.items():
            if wins == 0:
                res[0].append(num)
            elif wins == 1:
                res[1].append(num)
        res[1].sort()
        res[0].sort()
        return res
